Kernel/Storage/NVMe/NVMeDefinitions.h

```diff
@@ -160,3 +154,64 @@ struct NVMeSubmission {
     LittleEndian<u32> cdw14;
     LittleEndian<u32> cdw15;
 };
+
+struct [[gnu::packed]] NVMeRWCmd {
+    LittleEndian<u32> nsid;
+    LittleEndian<u64> rsvd;
+    LittleEndian<u64> metadata;
+    struct DataPtr data_ptr;
+    LittleEndian<u64> slba;
+    LittleEndian<u16> length;
+    LittleEndian<u16> control;
+    LittleEndian<u32> dsmgmt;
+    LittleEndian<u32> reftag;
+    LittleEndian<u16> apptag;
+    LittleEndian<u16> appmask;
+};
+
+struct [[gnu::packed]] NVMeIdentifyCmd {
+    LittleEndian<u32> nsid;
+    LittleEndian<u64> rsvd1[2];
+    struct DataPtr data_ptr;
+    u8 cns;
+    u8 rsvd2;
+    LittleEndian<u16> ctrlid;
+    u8 rsvd3[3];
+    u8 csi;
+    u64 rsvd4[2];
+};
+
+struct [[gnu::packed]] NVMeCreateCQCmd {
+    u32 rsvd1[5];
+    LittleEndian<u64> prp1;
+    u64 rsvd2;
+    LittleEndian<u16> cqid;
+    LittleEndian<u16> qsize;
+    LittleEndian<u16> cq_flags;
+    LittleEndian<u16> irq_vector;
+    u64 rsvd12[2];
+};
+
+struct [[gnu::packed]] NVMeCreateSQCmd {
+    u32 rsvd1[5];
+    LittleEndian<u64> prp1;
+    u64 rsvd2;
+    LittleEndian<u16> sqid;
+    LittleEndian<u16> qsize;
+    LittleEndian<u16> sq_flags;
+    LittleEndian<u16> cqid;
+    u64 rsvd12[2];
+};
+
+struct NVMeSubmission {
```

Comment:
Shouldn't this be marked `[[gnu::packed]]` as well?

Reply:
I tried first that but the compiler ignores the `[[gnu::packed]]` attribute as we have a union. So I don't think it makes sense to have them. 
```
NVMeDefinitions.h:210:5: error: ignoring packed attribute because of unpacked non-POD field 'NVMeSubmission::<unnamed union> NVMeSubmission::<anonymous>'
```

Reply:
Clang (https://godbolt.org/z/b3n459sae) doesn't emit a warning but gcc does (https://godbolt.org/z/Ys73x4dj3).

Reply:
You just need to declare the anonymous union as packed as well: https://godbolt.org/z/4r3hhE8d8

Reply:
Somehow my primitive brain thought that it was not possible for the compiler to pack a union. Lol. I will fix it. Thanks!